homeassistant/util/__init__.py

```diff
@@ -260,25 +285,21 @@ def wrapper(*args, **kwargs):
             else:
                 host = args[0] if args else wrapper
 
-            if not hasattr(host, '_throttle_lock'):
-                host._throttle_lock = threading.Lock()
+            if not hasattr(host, '_throttle'):
+                host._throttle = {}
 
-            if not host._throttle_lock.acquire(False):
-                return None
+            if not id(self) in host._throttle:
+                host._throttle[id(self)] = ThrottleItem()
```

Comment:
I don't think that this overhead is anything to be concerned about, and the code is easier to read when using an object. But if you rather see a list I can change